mat/doc.go

```diff
@@ -36,10 +36,25 @@
 //  zero.Copy(a)
 //
 // Receivers must be the correct size for the matrix operations, otherwise the
-// operation will panic. As a special case for convenience, a zero-sized matrix
+// operation will panic. As a special case for convenience, a zero-value matrix
 // will be modified to have the correct size, allocating data if necessary.
 //  var c mat.Dense // construct a new zero-sized matrix
-//  c.Mul(a, a)       // c is automatically adjusted to be 6×6
+//  c.Mul(a, a)     // c is automatically adjusted to be 6×6
+//
+// Zero-value of a matrix
+//
+// A zero-value matrix is either the Go language definition of a zero-value or
+// a matrix that has had its Reset method called. Matrix implementations provided
```

Comment:
That doesn't quite say what I want though; only resetable matrices have `IsZero`. At the moment, this is all of them, but `Band` won't in the short term, and it may be possible that we will provide non-resetable types in the future.